src/t_string.c

```diff
@@ -228,14 +228,14 @@ int parseExtendedStringArgumentsOrReply(client *c, int *flags, int *unit, robj *
         } else if (!strcasecmp(opt,"PERSIST") && (command_type == COMMAND_GET) &&
                !(*flags & OBJ_EX) && !(*flags & OBJ_EXAT) &&
                !(*flags & OBJ_PX) && !(*flags & OBJ_PXAT) &&
-               !(*flags & OBJ_KEEPTTL))
+               !(*flags & OBJ_KEEPTTL) && !(*flags & OBJ_PERSIST))
         {
             *flags |= OBJ_PERSIST;
         } else if ((opt[0] == 'e' || opt[0] == 'E') &&
                    (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
                    !(*flags & OBJ_KEEPTTL) && !(*flags & OBJ_PERSIST) &&
                    !(*flags & OBJ_EXAT) && !(*flags & OBJ_PX) &&
-                   !(*flags & OBJ_PXAT) && next)
+                   !(*flags & OBJ_PXAT) && next && !(*flags & OBJ_EX))
```

Comment:
(Minor unimportant suggestion) Since all these options are mutually exclusive, they can be listed in the same order in every else-if branch. `next` is different so I think it can be on a separate line. The flags could be checked all at once if you want:

```C
        } else if ((opt[0] == 'e' || opt[0] == 'E') &&
                   (opt[1] == 'x' || opt[1] == 'X') && opt[2] == '\0' &&
                   !(*flags & (OBJ_KEEPTTL | OBJ_PERSIST | OBJ_EX | OBJ_PX | OBJ_EXAT | OBJ_PXAT)) &&
                   next)
```

Reply:
Hello, thanks for the idea. I like this change as it will simplify the if statements. I have updated the code.
Thanks